libraries/WiFi/src/WiFiClient.h

```diff
@@ -20,30 +20,21 @@
 #ifndef _WIFICLIENT_H_
 #define _WIFICLIENT_H_
 
+
 #include "Arduino.h"
 #include "Client.h"
+#include <memory>
 
```

Comment:
You can move the whole class in the CPP and only forward declare it here. It's private really and users do not need to bother with it.